crates/libs/core/src/types/client/health.rs

```diff
@@ -0,0 +1,947 @@
+                    as *const FABRIC_STATELESS_SERVICE_INSTANCE_HEALTH_REPORT
+                    as *mut c_void,
+            },
+            HealthReport::Partition(report) => Self {
+                Kind: FABRIC_HEALTH_REPORT_KIND_PARTITION,
+                Value: &FABRIC_PARTITION_HEALTH_REPORT::from(report)
+                    as *const FABRIC_PARTITION_HEALTH_REPORT as *mut c_void,
+            },
+            HealthReport::Node(report) => Self {
+                Kind: FABRIC_HEALTH_REPORT_KIND_NODE,
+                Value: &FABRIC_NODE_HEALTH_REPORT::from(report) as *const FABRIC_NODE_HEALTH_REPORT
+                    as *mut c_void,
+            },
+            HealthReport::Service(report) => Self {
+                Kind: FABRIC_HEALTH_REPORT_KIND_SERVICE,
+                Value: &FABRIC_SERVICE_HEALTH_REPORT::from(report)
+                    as *const FABRIC_SERVICE_HEALTH_REPORT as *mut c_void,
+            },
+            HealthReport::Application(report) => Self {
+                Kind: FABRIC_HEALTH_REPORT_KIND_APPLICATION,
+                Value: &FABRIC_APPLICATION_HEALTH_REPORT::from(report)
+                    as *const FABRIC_APPLICATION_HEALTH_REPORT
+                    as *mut c_void,
+            },
+            HealthReport::DeployedApplication(report) => Self {
+                Kind: FABRIC_HEALTH_REPORT_KIND_DEPLOYED_APPLICATION,
+                Value: &FABRIC_DEPLOYED_APPLICATION_HEALTH_REPORT::from(report)
+                    as *const FABRIC_DEPLOYED_APPLICATION_HEALTH_REPORT
+                    as *mut c_void,
+            },
+            HealthReport::DeployedServicePackage(report) => Self {
+                Kind: FABRIC_HEALTH_REPORT_KIND_DEPLOYED_SERVICE_PACKAGE,
+                Value: &FABRIC_DEPLOYED_SERVICE_PACKAGE_HEALTH_REPORT::from(report)
+                    as *const FABRIC_DEPLOYED_SERVICE_PACKAGE_HEALTH_REPORT
+                    as *mut c_void,
+            },
+            HealthReport::Cluster(report) => Self {
+                Kind: FABRIC_HEALTH_REPORT_KIND_CLUSTER,
+                Value: &FABRIC_CLUSTER_HEALTH_REPORT::from(report)
+                    as *const FABRIC_CLUSTER_HEALTH_REPORT as *mut c_void,
+            },
+        }
+    }
+}
+
+/// FABRIC_STATEFUL_SERVICE_REPLICA_HEALTH_REPORT
+#[derive(Debug, Clone)]
+pub struct StatefulServiceReplicaHealthReport {
+    pub partition_id: GUID,
+    pub replica_id: i64,
+    pub health_information: HealthInformation,
+    // TODO: Implement reserved fields
+    // pub reserved: *mut std::ffi::c_void,
+}
+
+impl From<&StatefulServiceReplicaHealthReport> for FABRIC_STATEFUL_SERVICE_REPLICA_HEALTH_REPORT {
+    fn from(value: &StatefulServiceReplicaHealthReport) -> Self {
+        Self {
+            PartitionId: value.partition_id,
+            ReplicaId: value.replica_id,
+            HealthInformation: &FABRIC_HEALTH_INFORMATION::from(&value.health_information),
```

Comment:
This HealthInformation field is likely a dangling ptr.

Reply:
The `HealthInformation` field should have its lifetime equal to the lifetime of `value.health_information`. Could you elaborate on your concern?

Reply:
https://github.com/Azure/service-fabric-rs/blob/d9e9492adfd5ab36f25012592fe7d2f78e88627f/crates/libs/com/src/ServiceFabric/FabricTypes.rs#L12313

pub HealthInformation: *const FABRIC_HEALTH_INFORMATION, 
HealthInformation is a raw ptr. Is the raw ptr allocated on the heap? If not it is a ptr on local stack and thus dangling after returning.

Reply:
Fixed using `Box` for heap allocation.